### ISSUE TITLE
[jsx-filename-extension]: option to allow empty files

### ISSUE THREAD

**Issue by silverwind** [jsx-eslint/eslint-plugin-react#3644]
Given the config

```yaml
rules:
  react/jsx-filename-extension: [2, {allow: as-needed]}]
```

a file that is empty or contains only comments triggers the rule, for example a `.jsx` file with:

```js
// Put your components here
```

I think a option to allow empty files when `as-needed` is specified would be ideal for this use case, not sure whether it should be the default, but I think it could be.

**Comment by ljharb** [comment]
Why? That would mean that as soon as you add code to it, you might get an error that forces you to rename it.

**Comment by ljharb** [comment]
ohhh, i see - you *want* it to be `.jsx` but the rule is saying it's not needed and warning on it?

**Comment by silverwind** [comment]
Yes exactly. A empty file is kind of an edge-case for such on-demand `.jsx` extension.

**Comment by ljharb** [comment]
then yes, i think an additional option that ignores empty files would be reasonable.

**Comment by silverwind** [comment]
Okay, maybe I'll look into implementing it if it's not too hard.
